// hview-plus/es/src/message/src/message.mjs
import { defineComponent as k, toRefs as T, ref as B, computed as L, onMounted as M, onUnmounted as $, openBlock as o, createBlock as w, Transition as H, unref as e, withCtx as S, withDirectives as D, createElementVNode as m, normalizeClass as u, renderSlot as E, createElementBlock as n, toDisplayString as N, createCommentVNode as U, vShow as V } from "vue";
import { MessageProps as b } from "./message2.mjs";
import "../style/index.mjs";
const x = {
  key: 0,
  class: "h-message__content"
}, z = ["innerHTML"], A = {
  name: "HMessage"
}, F = /* @__PURE__ */ k({
  ...A,
  props: b,
  setup(f) {
    const d = f, { message: a, showClose: p, type: s, duration: l, center: _, dangerouslyUseHTMLString: v, onClose: h } = T(d);
    let t = B(!1), r = null;
    const g = L(() => s.value ? `h-message__icon h-icon-${s.value}` : ""), y = () => {
      t.value = !0, l.value > 0 && (r = setTimeout(() => {
        t.value = !1;
      }, l.value));
    }, C = () => {
      t.value = !1;
    };
    return M(() => {
      y();
    }), $(() => {
      clearTimeout(r);
    }), (i, c) => (o(), w(H, {
      name: "h-message-fade",
      onBeforeLeave: e(h),
      onAfterLeave: c[0] || (c[0] = (P) => i.$emit("destroy"))
    }, {
      default: S(() => [
        D(m("div", {
          class: u(["h-message", e(s) ? `h-message--${e(s)}` : "", e(_) ? "is-center" : ""])
        }, [
          m("i", {
            class: u(e(g))
          }, null, 2),
          E(i.$slots, "default", {}, () => [
            e(v) ? (o(), n("p", {
              key: 1,
              innerHTML: e(a),
              class: "h-message__content"
            }, null, 8, z)) : (o(), n("p", x, N(e(a)), 1))
          ]),
          e(p) ? (o(), n("i", {
            key: 0,
            class: "h-message__closeBtn iconfont h-icon-close",
            onClick: C
          })) : U("", !0)
        ], 2), [
          [V, e(t)]
        ])
      ]),
      _: 3
    }, 8, ["onBeforeLeave"]));
  }
});
export {
  F as default
};
