// hview-plus/es/src/carousel-item/src/carousel-item.mjs
import { defineComponent as y, computed as a, unref as e, withDirectives as _, openBlock as S, createElementBlock as I, normalizeClass as $, normalizeStyle as g, renderSlot as k, vShow as w } from "vue";
import "../style/index.mjs";
import { carouselItemProps as x } from "./carousel-item2.mjs";
import { useCarouselItem as z } from "./utils.mjs";
const B = {
  name: "HCarouselItem"
}, P = /* @__PURE__ */ y({
  ...B,
  props: x,
  setup(n) {
    const l = n, { active: i, animating: c, hover: m, inStage: u, isVertical: p, translate: f, scale: d, ready: h, handleItemClick: s } = z(
      l,
      "HCarouselItem"
    ), C = a(() => ({
      ["h-carousel-item--active"]: e(i),
      ["h-carousel-item--inStage"]: e(u),
      ["h-carousel-item--hover"]: e(m),
      ["h-carousel-item--animating"]: e(c)
    })), v = a(() => {
      const t = `${`translate${e(p) ? "Y" : "X"}`}(${e(f)}px)`, r = `scale(${e(d)})`;
      return {
        transform: [t, r].join(" ")
      };
    });
    return (o, t) => _((S(), I("div", {
      class: $(["h-carousel-item", e(C)]),
      style: g(e(v)),
      onClick: t[0] || (t[0] = (...r) => e(s) && e(s)(...r))
    }, [
      k(o.$slots, "default")
    ], 6)), [
      [w, e(h)]
    ]);
  }
});
export {
  P as default
};
