// hview-plus/es/src/avatar/src/avatar.mjs
import { defineComponent as d, ref as v, computed as o, watch as y, resolveComponent as z, openBlock as s, createElementBlock as n, normalizeClass as b, unref as a, normalizeStyle as l, createBlock as k, renderSlot as S } from "vue";
import "../style/index.mjs";
import { avatarProps as E, avatarEmits as _ } from "./avatar2.mjs";
const C = ["src", "alt", "srcset"], $ = {
  name: "HAvatar"
}, A = /* @__PURE__ */ d({
  ...$,
  props: E,
  emits: _,
  setup(c, { emit: i }) {
    const r = c, t = v(!1), p = o(() => ({
      [`h-avatar-size--${r.size}`]: typeof r.size != "number",
      [`h-avatar--${r.shape}`]: r.shape,
      [`h-avatar-border--${r.border}`]: r.border
    })), m = o(() => {
      const { size: e } = r;
      return typeof e == "number" ? { width: e + "px", height: e + "px" } : {};
    }), u = o(() => ({
      objectFit: r.fit
    }));
    y(
      () => r.src,
      () => t.value = !1
    );
    function f(e) {
      t.value = !0, i("error", e);
    }
    return (e, B) => {
      const h = z("h-icon");
      return s(), n("span", {
        class: b(["h-avatar", a(p)]),
        style: l(a(m))
      }, [
        (e.src || e.srcSet) && !t.value ? (s(), n("img", {
          key: 0,
          src: e.src,
          alt: e.alt,
          srcset: e.srcSet,
          style: l(a(u)),
          onError: f
        }, null, 44, C)) : e.icon ? (s(), k(h, {
          key: 1,
          name: e.icon
        }, null, 8, ["name"])) : S(e.$slots, "default", { key: 2 })
      ], 6);
    };
  }
});
export {
  A as default
};
