// hview-plus/es/src/tree/src/transition.mjs
import { defineComponent as d, openBlock as p, createBlock as l, Transition as c, withCtx as u, renderSlot as O } from "vue";
import g from "../../../_virtual/plugin-vue_export-helper.mjs";
const h = d({
  name: "HCollapseTransition",
  props: {
    duration: {
      type: Number,
      default: 0.5
    },
    animate: {
      type: String,
      default: "ease"
    }
  },
  setup(t) {
    const o = (e) => e.currentStyle ? e.currentStyle : e.currentStyle || getComputedStyle(e), n = (e, a) => {
      if (!!e)
        for (const i in a)
          Object.prototype.hasOwnProperty.call(a, i) && (e.style[i] = a[i]);
    }, r = {
      transition: `all ${t.duration}s ${t.animate}`,
      overflow: "hidden"
    };
    return {
      OnbeforeEnter: (e) => {
        e.dataset.paddingTop = o(e).paddingTop, e.dataset.paddingBottom = o(e).paddingBottom, n(e, Object.assign(r, { height: 0 }));
      },
      Onenter: (e) => {
        setTimeout(() => {
          n(e, {
            height: `${e.scrollHeight}px`,
            paddingTop: e.dataset.paddingTop,
            paddingBottom: e.dataset.paddingBottom
          });
        }, 0);
      },
      OnafterEnter: (e) => {
        n(e, { height: "", overflow: "" });
      },
      OnbeforeLeave: (e) => {
        n(
          e,
          Object.assign(r, {
            height: `${e.scrollHeight}px`,
            overflow: ""
          })
        );
      },
      Onleave: (e) => {
        setTimeout(() => {
          n(e, { height: 0, overflow: "hidden" });
        }, 0);
      },
      OnafterLeave: (e) => {
        n(e, { height: 0, overflow: "hidden" });
      }
    };
  }
});
function v(t, o, n, r, f, s) {
  return p(), l(c, {
    onBeforeEnter: t.OnbeforeEnter,
    onEnter: t.Onenter,
    onAfterEnter: t.OnafterEnter,
    onBeforeLeave: t.OnbeforeLeave,
    onLeave: t.Onleave,
    onAfterLeave: t.OnafterLeave
  }, {
    default: u(() => [
      O(t.$slots, "default")
    ]),
    _: 3
  }, 8, ["onBeforeEnter", "onEnter", "onAfterEnter", "onBeforeLeave", "onLeave", "onAfterLeave"]);
}
const T = /* @__PURE__ */ g(h, [["render", v]]);
export {
  T as default
};
