// hview-plus/es/src/tag/src/tag.mjs
import { defineComponent as y, computed as b, unref as t, resolveComponent as $, openBlock as s, createElementBlock as z, normalizeClass as m, normalizeStyle as d, createElementVNode as n, renderSlot as h, createBlock as l, withModifiers as u, createCommentVNode as k, Transition as B, withCtx as E } from "vue";
import "../style/index.mjs";
import { tagProps as T, tagEmits as w } from "./tag2.mjs";
const N = { class: "h-tag-content" }, S = { class: "h-tag-content" }, V = {
  name: "HTag"
}, P = /* @__PURE__ */ y({
  ...V,
  props: T,
  emits: w,
  setup(g, { emit: a }) {
    const C = g, c = b(() => {
      const { type: o, border: f, theme: e, round: _, size: p } = C;
      return {
        ["h-tag-round"]: _,
        [`h-tag--${o}`]: t(o),
        [`h-tag--${p}`]: t(p),
        [`is-${e}`]: t(e)
      };
    }), r = (o) => {
      a("close", o);
    }, i = (o) => {
      a("click", o);
    };
    return (o, f) => {
      const e = $("h-icon");
      return o.disableTransitions ? (s(), z("span", {
        key: 0,
        class: m(["h-tag", t(c)]),
        style: d({ backgroundColor: o.color }),
        onClick: i
      }, [
        n("span", N, [
          h(o.$slots, "default")
        ]),
        o.closable ? (s(), l(e, {
          key: 0,
          name: "close",
          class: "h-tag-close",
          onClick: u(r, ["stop"])
        }, null, 8, ["onClick"])) : k("", !0)
      ], 6)) : (s(), l(B, {
        key: 1,
        name: "h-tag-zoom-in-center",
        appear: ""
      }, {
        default: E(() => [
          n("span", {
            class: m(["h-tag", t(c)]),
            style: d({ backgroundColor: o.color }),
            onClick: i
          }, [
            n("span", S, [
              h(o.$slots, "default")
            ]),
            o.closable ? (s(), l(e, {
              key: 0,
              name: "close",
              class: "h-tag-close",
              onClick: u(r, ["stop"])
            }, null, 8, ["onClick"])) : k("", !0)
          ], 6)
        ]),
        _: 3
      }));
    };
  }
});
export {
  P as default
};
