// hview-plus/es/src/modal/src/modal.mjs
import { defineComponent as m, openBlock as o, createBlock as b, Transition as u, withCtx as h, createElementBlock as t, normalizeStyle as p, createElementVNode as l, renderSlot as d, createTextVNode as r, createCommentVNode as a } from "vue";
import "../style/index.mjs";
import { modalProps as f } from "./modal2.mjs";
const C = {
  class: "modal",
  role: "dialog",
  "aria-labelledby": "modalTitle",
  "aria-describedby": "modalDescription"
}, y = {
  key: 0,
  class: "modal-header",
  id: "modalTitle"
}, k = {
  class: "modal-body",
  id: "modalDescription"
}, _ = {
  key: 1,
  class: "modal-footer"
}, T = {
  name: "HModal"
}, N = /* @__PURE__ */ m({
  ...T,
  props: f,
  emits: ["onClose", "update:show"],
  setup(c, { emit: n }) {
    const i = c, s = () => {
      n("update:show", !i.show), n("onClose");
    };
    return (e, w) => (o(), b(u, { name: "modal-fade" }, {
      default: h(() => [
        i.show ? (o(), t("div", {
          key: 0,
          class: "modal-backdrop",
          style: p({ zIndex: e.zIndex, backgroundColor: e.bgColor })
        }, [
          l("div", C, [
            e.title ? (o(), t("header", y, [
              d(e.$slots, "header", {}, () => [
                r(" This is the default title! ")
              ]),
              l("button", {
                type: "button",
                class: "btn-close",
                onClick: s,
                "aria-label": "Close modal"
              }, " x ")
            ])) : a("", !0),
            l("section", k, [
              !e.title && !e.footer ? (o(), t("button", {
                key: 0,
                type: "button",
                class: "btn-close",
                onClick: s,
                "aria-label": "Close modal"
              }, " x ")) : a("", !0),
              d(e.$slots, "body", {}, () => [
                r(" This is the default body! ")
              ])
            ]),
            e.footer ? (o(), t("footer", _, [
              d(e.$slots, "footer", {}, () => [
                r(" This is the default footer! ")
              ]),
              l("button", {
                type: "button",
                class: "btn-green",
                onClick: s,
                "aria-label": "Close modal"
              }, " Close Modal ")
            ])) : a("", !0)
          ])
        ], 4)) : a("", !0)
      ]),
      _: 3
    }));
  }
});
export {
  N as default
};
