// hview-plus/es/src/radio/src/radio.mjs
import { defineComponent as g, ref as c, openBlock as r, createElementBlock as a, Fragment as v, renderList as y, normalizeClass as d, normalizeStyle as u, createElementVNode as k, createTextVNode as C, toDisplayString as b } from "vue";
import "../style/index.mjs";
import { radioProps as x } from "./radio2.mjs";
const S = { class: "h-radio" }, R = ["onClick"], z = {
  name: "HRadio"
}, V = /* @__PURE__ */ g({
  ...z,
  props: x,
  emits: ["change"],
  setup(p, { emit: m }) {
    const s = p, n = c(-1), t = c("inline-flex"), f = () => (s.inline ? t.value : t.value = "flex", { display: `${t.value}` }), h = (e) => n.value == e ? { marginRight: "4px", backgroundColor: s.customColor } : { marginRight: "4px" }, _ = (e, i) => {
      e.disabled || (n.value = i), m("change", e);
    };
    return (e, i) => (r(), a("div", S, [
      (r(!0), a(v, null, y(e.options, (l, o) => (r(), a("div", {
        class: d(["h-item-radio", { "h-item-disabled-radio": l.disabled }]),
        key: o,
        style: u(f())
      }, [
        k("span", {
          style: u(h(o)),
          class: d({ active: n.value == o }),
          onClick: (B) => _(l, o)
        }, null, 14, R),
        C(b(l.lable), 1)
      ], 6))), 128))
    ]));
  }
});
export {
  V as default
};
