// hview-plus/es/src/checkbox/src/checkbox.mjs
import { defineComponent as _, inject as f, computed as n, openBlock as h, createElementBlock as p, normalizeClass as b, unref as s, createElementVNode as c, withDirectives as g, isRef as x, vModelCheckbox as C, renderSlot as V, Fragment as y, createTextVNode as B, toDisplayString as D, createCommentVNode as E, nextTick as G } from "vue";
import { CheckBoxProps as N } from "./checkbox2.mjs";
import "../style/index.mjs";
const S = /* @__PURE__ */ c("span", { class: "h-checkbox__inner" }, null, -1), T = ["disabled", "value"], $ = { class: "h-checkbox__label" }, j = {
  name: "HCheckbox"
}, H = /* @__PURE__ */ _({
  ...j,
  props: N,
  emits: ["update:modelValue", "change"],
  setup(m, { emit: r }) {
    const a = m, l = f("checkboxGroup", {}), t = n(() => !!(l && l.modelValue)), o = n({
      get() {
        return t.value ? l && l.modelValue.value : a.modelValue;
      },
      set(e) {
        t.value ? l && l.changeEvent(e) : r("update:modelValue", e);
      }
    }), i = n(() => t.value ? o.value.includes(a.label) : o.value), d = n(() => t.value && l && l.disabled.value || a.disabled), k = () => {
      G(() => r("change", o.value));
    };
    return (e, u) => (h(), p("label", {
      class: b(["h-checkbox", [{ "is-checked": s(i) }, { "is-disabled": s(d) }, { "is-bordered": e.border }]])
    }, [
      c("span", {
        class: b(["h-checkbox__input", {
          "is-checked": s(i),
          "is-disabled": s(d),
          "is-indeterminate": e.indeterminate
        }])
      }, [
        S,
        g(c("input", {
          class: "h-checkbox__original",
          type: "checkbox",
          "onUpdate:modelValue": u[0] || (u[0] = (v) => x(o) ? o.value = v : null),
          disabled: s(d),
          value: e.label,
          onChange: k
        }, null, 40, T), [
          [C, s(o)]
        ])
      ], 2),
      c("span", $, [
        V(e.$slots, "default"),
        e.$slots.default ? E("", !0) : (h(), p(y, { key: 0 }, [
          B(D(e.label), 1)
        ], 64))
      ])
    ], 2));
  }
});
export {
  H as default
};
