// hview-plus/es/src/badge/src/badge.mjs
import { defineComponent as l, computed as s, openBlock as u, createElementBlock as p, renderSlot as c, withDirectives as d, createElementVNode as m, normalizeClass as f, unref as n, toDisplayString as _, vShow as h } from "vue";
import "../style/index.mjs";
import { badgeProps as v } from "./badge2.mjs";
const y = { class: "h-badge" }, D = ["textContent"], b = {
  name: "HBadge"
}, B = /* @__PURE__ */ l({
  ...b,
  props: v,
  setup(r, { expose: a }) {
    const e = r, i = s(() => e.type && e.value && !e.isDot ? {
      [`h-badge-type--${e.type}`]: e.type
    } : {
      "is-dot": e.isDot
    }), t = s(() => e.isDot ? "" : typeof e.value == "number" && typeof e.max == "number" ? e.max < e.value ? `${e.max}+` : `${e.value}` : `${e.value}`);
    return a({
      content: t
    }), (o, g) => (u(), p("div", y, [
      c(o.$slots, "default"),
      d(m("sup", {
        class: f(n(i)),
        textContent: _(n(t))
      }, null, 10, D), [
        [h, !o.hidden && (n(t) || o.isDot)]
      ])
    ]));
  }
});
export {
  B as default
};
