// hview-plus/es/index.mjs
import r from "./src/index.mjs";
import a from "@hview-plus/locale";
import { default as u } from "./src/button/index.mjs";
import { default as p } from "./src/link/index.mjs";
import { default as n } from "./src/icon/index.mjs";
import { default as c } from "./src/card/index.mjs";
import { default as g } from "./src/input/index.mjs";
import { Aside as k, Container as B, Footer as I, Header as M, Main as T } from "./src/container/index.mjs";
import { default as A } from "./src/avatar/index.mjs";
import { default as R } from "./src/radio/index.mjs";
import { default as v } from "./src/tooltip/index.mjs";
import { default as F } from "./src/carousel/index.mjs";
import { default as H } from "./src/carousel-item/index.mjs";
import { default as P } from "./src/message/src/index.mjs";
import { default as q } from "./src/scrollbar/index.mjs";
import { default as z } from "./src/tag/index.mjs";
import { default as J } from "./src/progress/index.mjs";
import { default as N } from "./src/result/index.mjs";
import { default as Q } from "./src/slider/index.mjs";
import { default as V } from "./src/tree/index.mjs";
import { default as X } from "./src/checkboxGroup/index.mjs";
import { default as Z } from "./src/checkbox/index.mjs";
import { default as $ } from "./src/badge/index.mjs";
import { default as eo } from "./src/modal/index.mjs";
import { default as ao } from "./src/upload/index.mjs";
import { default as fo } from "./src/empty/index.mjs";
const s = {
  install: (o) => {
    r.forEach((e) => o.use(e)), o.use(a);
  }
};
export {
  k as Aside,
  A as Avatar,
  $ as Badge,
  u as Button,
  c as Card,
  F as Carousel,
  H as CarouselItem,
  Z as Checkbox,
  X as CheckboxGroup,
  B as Container,
  fo as Empty,
  I as Footer,
  M as Header,
  n as Icon,
  g as Input,
  p as Link,
  T as Main,
  P as Message,
  eo as Modal,
  J as Progress,
  R as Radio,
  N as Result,
  q as ScrollBar,
  Q as Slider,
  z as Tag,
  v as Tooltip,
  V as Tree,
  s as default,
  ao as upload
};
