// hview-plus/es/src/carousel/src/carousel.mjs
import { defineComponent as K, computed as h, unref as e, resolveComponent as O, openBlock as n, createElementBlock as a, normalizeClass as d, withModifiers as l, createElementVNode as p, normalizeStyle as Q, createBlock as g, Transition as B, withCtx as E, withDirectives as I, createVNode as L, vShow as S, createCommentVNode as m, renderSlot as R, Fragment as U, renderList as W, toDisplayString as X } from "vue";
import "../style/index.mjs";
import { carouselProps as Y, carouselEmits as Z } from "./carousel2.mjs";
import { useCarousel as x } from "./utils.mjs";
const _ = ["onMouseenter", "onClick"], ee = { key: 0 }, oe = {
  name: "HCarousel"
}, ie = /* @__PURE__ */ K({
  ...oe,
  props: Y,
  emits: Z,
  setup(D, { expose: H, emit: N }) {
    const r = D, {
      root: P,
      activeIndex: s,
      arrowDisplay: v,
      hasLabel: f,
      hover: w,
      items: y,
      handleButtonEnter: C,
      handleButtonLeave: u,
      handleIndicatorClick: T,
      handleMouseEnter: k,
      handleMouseLeave: M,
      setActiveItem: V,
      prev: z,
      next: A,
      throttledArrowClick: $,
      throttledIndicatorHover: F
    } = x(r, N, "Hcarousel"), j = h(() => ({
      [`h-carousel--${r.direction}`]: r.direction
    })), q = h(() => ({
      [`h-carousel-indicators--${r.direction}`]: r.direction,
      ["h-carousel-indicators-labels"]: e(f),
      ["h-carousel-indicators-outside"]: r.indicatorPosition === "outside"
    })), G = h(() => ({
      [`h-carousel-indicator-button--${r.indicatorType}`]: r.indicatorType
    }));
    return H({
      setActiveItem: V,
      prev: z,
      next: A
    }), (i, o) => {
      const b = O("h-icon");
      return n(), a("div", {
        ref_key: "root",
        ref: P,
        class: d(["h-carousel", e(j)]),
        onMouseenter: o[6] || (o[6] = l(
          (...t) => e(k) && e(k)(...t),
          ["stop"]
        )),
        onMouseleave: o[7] || (o[7] = l(
          (...t) => e(M) && e(M)(...t),
          ["stop"]
        ))
      }, [
        p("div", {
          class: "h-carousel-container",
          style: Q({ height: i.height })
        }, [
          e(v) ? (n(), g(B, {
            key: 0,
            name: "carousel-arrow-left"
          }, {
            default: E(() => [
              I(p("button", {
                type: "button",
                class: "h-carousel-arrow h-carousel-arrow-left",
                onMouseenter: o[0] || (o[0] = (t) => e(C)("left")),
                onMouseleave: o[1] || (o[1] = (...t) => e(u) && e(u)(...t)),
                onClick: o[2] || (o[2] = l((t) => e($)(e(s) - 1), ["stop"]))
              }, [
                L(b, { name: "arrow-left" })
              ], 544), [
                [S, (i.arrow === "always" || e(w)) && (r.loop || e(s) > 0)]
              ])
            ]),
            _: 1
          })) : m("", !0),
          e(v) ? (n(), g(B, {
            key: 1,
            name: "carousel-arrow-right"
          }, {
            default: E(() => [
              I(p("button", {
                type: "button",
                class: "h-carousel-arrow h-carousel-arrow-right",
                onMouseenter: o[3] || (o[3] = (t) => e(C)("right")),
                onMouseleave: o[4] || (o[4] = (...t) => e(u) && e(u)(...t)),
                onClick: o[5] || (o[5] = l((t) => e($)(e(s) + 1), ["stop"]))
              }, [
                L(b, { name: "arrow-right" })
              ], 544), [
                [S, (i.arrow === "always" || e(w)) && (r.loop || e(s) < e(y).length - 1)]
              ])
            ]),
            _: 1
          })) : m("", !0),
          R(i.$slots, "default")
        ], 4),
        i.indicatorPosition !== "none" ? (n(), a("ul", {
          key: 0,
          class: d(["h-carousel-indicators", e(q)])
        }, [
          (n(!0), a(U, null, W(e(y), (t, c) => (n(), a("li", {
            key: c,
            class: d(["h-carousel-indicator", {
              "h-carousel-indicator-active": c === e(s),
              [`h-carousel-indicator--${r.direction}`]: r.direction
            }]),
            onMouseenter: (J) => e(F)(c),
            onClick: l((J) => e(T)(c), ["stop"])
          }, [
            p("button", {
              class: d(["h-carousel-indicator-button", e(G)])
            }, [
              e(f) ? (n(), a("span", ee, X(t.props.label), 1)) : m("", !0)
            ], 2)
          ], 42, _))), 128))
        ], 2)) : m("", !0)
      ], 34);
    };
  }
});
export {
  ie as default
};
