// hview-plus/es/src/carousel/src/carousel2.mjs
const o = {
  initialIndex: {
    type: Number,
    default: 0
  },
  height: {
    type: String,
    default: ""
  },
  trigger: {
    type: String,
    values: ["hover", "click"],
    default: "hover"
  },
  autoplay: {
    type: Boolean,
    default: !0
  },
  interval: {
    type: Number,
    default: 3e3
  },
  indicatorPosition: {
    type: String,
    values: ["", "none", "outside"],
    default: ""
  },
  indicator: {
    type: Boolean,
    default: !0
  },
  indicatorType: {
    type: String,
    values: ["circle", "rect"],
    default: "circle"
  },
  arrow: {
    type: String,
    values: ["always", "hover", "never"],
    default: "hover"
  },
  loop: {
    type: Boolean,
    default: !0
  },
  direction: {
    type: String,
    values: ["horizontal", "vertical"],
    default: "horizontal"
  },
  pauseOnHover: {
    type: Boolean,
    default: !0
  }
}, a = {
  change: (e, t) => [e, t].every((r) => Number.isInteger(r))
}, l = Symbol("carouselContextKey");
export {
  l as carouselContextKey,
  a as carouselEmits,
  o as carouselProps
};
