// hview-plus/es/src/carousel/src/utils.mjs
import { getCurrentInstance as te, ref as g, computed as N, unref as T, watch as m, shallowRef as k, onMounted as ne, onBeforeUnmount as oe, provide as le, isVNode as w } from "vue";
import { useResizeObserver as re } from "@vueuse/core";
import { carouselContextKey as ae } from "./carousel2.mjs";
import { isString as ue, isArray as O } from "@vue/shared";
const b = (l) => {
  const s = O(l) ? l : [l], r = [];
  return s.forEach((t) => {
    var v;
    O(t) ? r.push(...b(t)) : w(t) && O(t.children) ? r.push(...b(t.children)) : (r.push(t), w(t) && ((v = t.component) == null ? void 0 : v.subTree) && r.push(...b(t.component.subTree)));
  }), r;
}, se = (l, s, r) => b(l.subTree).filter(
  (a) => {
    var n;
    return w(a) && ((n = a.type) == null ? void 0 : n.name) === s && !!a.component;
  }
).map((a) => a.component.uid).map((a) => r[a]).filter((a) => !!a), ie = (l, s) => {
  const r = {}, t = k([]);
  return {
    children: t,
    addChild: (n) => {
      r[n.uid] = n, t.value = se(l, s, r);
    },
    removeChild: (n) => {
      delete r[n], t.value = t.value.filter((i) => i.uid !== n);
    }
  };
}, R = 300, de = (l, s, r) => {
  const {
    children: t,
    addChild: v,
    removeChild: a
  } = ie(te(), "HCarouselItem"), n = g(-1), i = g(null), E = g(!1), d = g(), z = N(() => l.arrow !== "never" && !T(h)), P = N(() => t.value.some((e) => e.props.label.toString().length > 0)), h = N(() => l.direction === "vertical"), V = (e) => {
    let o = null;
    o || (o = setTimeout(() => {
      c(e), o = null;
    }, R));
  }, _ = (e) => {
    let o = null;
    o || (o = setTimeout(() => {
      Q(e), o = null;
    }, R));
  };
  function I() {
    i.value && (clearInterval(i.value), i.value = null);
  }
  function C() {
    l.interval <= 0 || !l.autoplay || i.value || (i.value = setInterval(() => D(), l.interval));
  }
  const D = () => {
    n.value < t.value.length - 1 ? n.value = n.value + 1 : l.loop && (n.value = 0);
  };
  function c(e) {
    if (ue(e)) {
      const f = t.value.filter((A) => A.props.name === e);
      f.length > 0 && (e = t.value.indexOf(f[0]));
    }
    if (e = Number(e), Number.isNaN(e) || e !== Math.floor(e)) {
      console.error(r, "index\u5FC5\u987B\u662F\u6570\u5B57");
      return;
    }
    const o = t.value.length, u = n.value;
    e < 0 ? n.value = l.loop ? o - 1 : 0 : e >= o ? n.value = l.loop ? 0 : o - 1 : n.value = e, u === n.value && p(u), B();
  }
  function p(e) {
    t.value.forEach((o, u) => {
      o.translateItem(u, n.value, e);
    });
  }
  function K(e, o) {
    var L, M, S, F;
    const u = T(t), f = u.length;
    if (f === 0 || !e.states.inStage)
      return !1;
    const A = o + 1, Y = o - 1, H = f - 1, Z = u[H].states.active, $ = u[0].states.active, x = (M = (L = u[A]) == null ? void 0 : L.states) == null ? void 0 : M.active, ee = (F = (S = u[Y]) == null ? void 0 : S.states) == null ? void 0 : F.active;
    return o === H && $ || x ? "left" : o === 0 && Z || ee ? "right" : !1;
  }
  function U() {
    E.value = !0, l.pauseOnHover && I();
  }
  function j() {
    E.value = !1, C();
  }
  function q(e) {
    T(h) || t.value.forEach((o, u) => {
      e === K(o, u) && (o.states.hover = !0);
    });
  }
  function G() {
    T(h) || t.value.forEach((e) => {
      e.states.hover = !1;
    });
  }
  function J(e) {
    n.value = e;
  }
  function Q(e) {
    l.trigger === "hover" && e !== n.value && (n.value = e);
  }
  function W() {
    c(n.value - 1);
  }
  function X() {
    c(n.value + 1);
  }
  function B() {
    I(), C();
  }
  m(
    () => n.value,
    (e, o) => {
      p(o), o > -1 && s("change", e, o);
    }
  ), m(
    () => l.autoplay,
    (e) => {
      e ? C() : I();
    }
  ), m(
    () => l.loop,
    () => {
      c(n.value);
    }
  ), m(
    () => l.interval,
    () => {
      B();
    }
  ), m(
    () => t.value,
    () => {
      t.value.length > 0 && c(l.initialIndex);
    }
  );
  const y = k();
  return ne(() => {
    y.value = re(d.value, () => {
      p();
    }), C();
  }), oe(() => {
    I(), d.value && y.value && y.value.stop();
  }), le(ae, {
    root: d,
    isVertical: h,
    items: t,
    loop: l.loop,
    addItem: v,
    removeItem: a,
    setActiveItem: c
  }), {
    root: d,
    activeIndex: n,
    arrowDisplay: z,
    hasLabel: P,
    hover: E,
    items: t,
    handleButtonEnter: q,
    handleButtonLeave: G,
    handleIndicatorClick: J,
    handleMouseEnter: U,
    handleMouseLeave: j,
    setActiveItem: c,
    prev: W,
    next: X,
    throttledArrowClick: V,
    throttledIndicatorHover: _
  };
};
export {
  b as flattedChildren,
  de as useCarousel,
  ie as useOrderedChildren
};
