// hview-plus/es/src/empty/src/empty.mjs
import { defineComponent as l, computed as r, openBlock as t, createElementBlock as o, createElementVNode as m, normalizeStyle as c, unref as n, renderSlot as i, createVNode as y, toDisplayString as u, createCommentVNode as f } from "vue";
import "../style/index.mjs";
import g from "./empty-img.mjs";
import { emptyProps as h } from "./empty2.mjs";
const _ = { class: "h-empty" }, k = ["src"], S = { class: "h-empty-description" }, $ = { key: 1 }, v = {
  key: 0,
  class: "h-empty-bottom"
}, E = {
  name: "HEmpty"
}, D = /* @__PURE__ */ l({
  ...E,
  props: h,
  setup(a) {
    const s = a, p = r(() => s.description || "description"), d = r(() => ({
      width: s.imageSize ? `${s.imageSize}px` : ""
    }));
    return (e, N) => (t(), o("div", _, [
      m("div", {
        class: "h-empty-image",
        style: c(n(d))
      }, [
        e.image ? (t(), o("img", {
          key: 0,
          src: e.image,
          ondragstart: "return false"
        }, null, 8, k)) : i(e.$slots, "image", { key: 1 }, () => [
          y(g)
        ])
      ], 4),
      m("div", S, [
        e.$slots.description ? i(e.$slots, "No data", { key: 0 }) : (t(), o("p", $, u(n(p)), 1))
      ]),
      e.$slots.default ? (t(), o("div", v, [
        i(e.$slots, "default")
      ])) : f("", !0)
    ]));
  }
});
export {
  D as default
};
